### DISCUSSION TITLE
Logseq SimpRead Sync 一站式标注及管理解决方案

### DISCUSSION THREAD

**Discussion by Kenshin** [Kenshin/simpread#4117] [服务]
![Logseq Sync](https://user-images.githubusercontent.com/81074/176084816-81304cde-4e3e-4522-93fe-78a08eda0db8.png)

# 截图

![image](https://user-images.githubusercontent.com/81074/175770233-7f807907-92ae-46e7-afb5-615890927a51.png)

# 视频

https://user-images.githubusercontent.com/81074/175770485-06c4f48d-68ef-40a2-a21c-c9351b2a60c6.mp4

# 描述

将简悦的标注与稍后读集成到 Logseq 进而实现：**集文章标注实时同步到 Logseq、本地快照实时保存到本地、管理你全部的稍后读于一身的 Logseq + 简悦全流程化一站式解决方案**。

# 下载地址

此功能内置到了 同步助手 1.1.1 版 [下载地址](http://ksria.com/simpread/docs/#/Sync?id=%e4%b8%8b%e8%bd%bd)。

# 优势

- 内置与 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725) 具有一样的模板解析引擎。

- 在 Web 标注 → 标注的 Markdown 自动出现在你的 Logseq。（全自动化方案）

- 标注的页面也可以同步生成本地永久快照。

- 每条标注均可链接到本地永久快照的原文对应位置。

- 得益于简悦精准的阅读模式匹配功能以及 HTML → Markdown 方案，通过此方式生成的 Markdown 的精准度最高，甚至可以匹配数学公式（LaTeX）

# 集成标注功能

得益于 [同步助手 1.1.1](https://github.com/Kenshin/simpread/discussions/4049#discussioncomment-2982511) 的 [标注嵌入模式](https://github.com/Kenshin/simpread/discussions/4070)。

终于实现**在 Logseq 中标注本地快照并实时同步到你的 Logseq，不仅如此还可以管理你的稍后读**。

# 受众

如果你是简悦 + Logseq 的深度使用者，尤其是使用同步助手的用户，非常建议使用此方式，会节省你大量的时间。

# 一站式教程

https://www.yuque.com/kenshin/simpread/gbere7

# 使用前提

此插件需要使用 [简悦 · 同步助手](http://ksria.com/simpread/docs/#/Sync)，才可发挥最大的效果，如何配置同步助手 [请看这里](https://kb.simpread.pro/#/page/配置同步助手)，同时也需要 [配置知识库](https://kb.simpread.pro/#/page/建立知识库)。

# 如何使用

https://github.com/Kenshin/simpread/discussions/4117#discussioncomment-3021599

# 设置

https://github.com/Kenshin/simpread/discussions/4117#discussioncomment-3021559

# 模板

因为内置了 [Markdown 模板辅助增强](https://github.com/Kenshin/simpread/discussions/3725) 一样的模板，所以 [适用于 Logseq 的模板](https://github.com/Kenshin/simpread/discussions/3725#discussioncomment-2635600) 都可以使用。

# 插件

基于此流程简悦官方推出了一个插件，可以方便的将部分手动功能改为自动化功能，细节请看 https://github.com/Kenshin/simpread/discussions/4220

# 注意

- 此方式不是插件方案，所以能做的比 [Obsidian 版](https://github.com/Kenshin/simpread/discussions/2889) 少一些，但改有的功能一样都不少。

- 简悦稍后读 → Logseq 属于单向更新。

- 通过此插件生成的 Markdown 的任何改动都可能被覆盖。

- 在简悦删除稍后读仅能删除扩展端的稍后读，无法同步删除 Logseq 对应的文件。

# 关联

简悦与 Logseq 都是基于 Local first 概念，即使只是轻量级使用简悦，也可以实现很多联动：

- [自动导入 org-mode 格式的标注到 Logseq](https://github.com/Kenshin/simpread/discussions/3802)

- [利用 Keyboard Maestro 的文件夹监控功能，将简悦标记导入到 Logseq（Mac 方案）](https://github.com/Kenshin/simpread/discussions/3581)

- [简悦 + Logseq + Github Page 无代码全自动化知识管理发布方案](https://github.com/Kenshin/simpread/discussions/3426)

关于简悦与 Logseq 的更多联动 [请看这里](https://github.com/Kenshin/simpread/discussions?discussions_q=label%3Alogseq)。

**Comment by Kenshin** [comment]
# 设置

此功能因为不是插件方案，所以没有办法把选项内置到 Logseq 中（ Logseq 插件稍晚些推出），所以相关选项内置到了 **标注嵌入模式**

## 添加

在任意 `http://localhost:7026/unread/2448` 加上后缀  `?embed=logseq` 

![image](https://user-images.githubusercontent.com/81074/175762263-cd38dfa0-3252-40e6-ba00-909968f86985.png)

## 设置项

![image](https://user-images.githubusercontent.com/81074/175759803-837f380e-108a-400c-b9b0-8201f5a2eddf.png)

## 设置路径

详细说明 [请看这里](https://github.com/Kenshin/simpread/discussions/4117#discussioncomment-3029410)。

## 导出全部

可以将当前全部稍后读的数据都导出（可选 `.org` 或 `.md` 格式），此功能需要设定路径。

![image](https://user-images.githubusercontent.com/81074/176087044-4c4e1d1f-a4bd-4836-9230-742568b763c6.png)

**Comment by Kenshin** [comment]
# 最新提示

下面的手动方式已经集成到简悦的 Logseq 插件中，可直接安装 [此插件](https://github.com/Kenshin/simpread/discussions/4220)。

# 手动方案

## 步骤

1. 新建一个空白的页面

   ![image](https://user-images.githubusercontent.com/81074/175762944-470acda5-b309-4c48-bbe6-6298bb62d9d6.png)

2. 随便起个名字 e.g. **简悦稍后读 · 索引页**

   ![image](https://user-images.githubusercontent.com/81074/175762980-3ec6da13-84cb-4162-bb4a-137ddc42f01b.png)

3. 建立 `iframe`

   `<iframe src="http://localhost:7026/unread?embed=logseq" width="100%" height="xxxpx"/>`  其中 `xxx` 按照自身的尺寸设定即可。

   ![image](https://user-images.githubusercontent.com/81074/175763030-4055f449-b343-48ca-8a0f-069f054f35bb.png)

4. 设置完后会显示下图效果

![image](https://user-images.githubusercontent.com/81074/175763096-00f9eb07-a8e6-42fb-9ad3-ac3aaa825065.png)

5. 添加收藏

   ![image](https://user-images.githubusercontent.com/81074/175763182-8bbb85ef-d440-4326-8301-1020e6989126.png)

## 如何使用

因为添加了收藏，所以从侧栏就能看到

![image](https://user-images.githubusercontent.com/81074/175763200-67d98e0e-45dc-4b8f-9273-37c3581250d7.png)

为了实现类似 [Obsidian Plugin: SimpRead Unreader Sync](https://github.com/Kenshin/simpread/discussions/2889) 左右侧栏的效果，需要把 **简悦稍后读 · 索引页** 在右侧打开。

使用快捷键 <kbd>Alt</kbd> <kbd>Shift</kbd> + <kbd>鼠标左键</kbd> 打开

## 演示

https://user-images.githubusercontent.com/81074/175764387-073953ec-0165-4ce4-8317-001f00f4e149.mp4

**Comment by Kenshin** [comment]
# 设置路径

注意：这个嵌入模式的保存是通过 [同步助手 · 增强导出](http://ksria.com/simpread/docs/#/Sync?id=增强导出) 设定的，无法通过设置项设定。

![image](https://user-images.githubusercontent.com/81074/175887204-51d96173-1f13-491a-b0ee-697c06f32100.png)

将下面的代码放到上面增强导出这里。

```
{"extension":"org", "path":"/Users/xxxx/Documents/Logseq/pages"}
```

## 注意

1. `path` 的内容是绝对路径地址，而非相对路径
2. `extension` 的值包括：`org` 和 `md`，根据上面设置项的保存格式设定。

**Comment by Kenshin** [comment]
# 文件名

如无特殊要求，默认即可，不需要修改。

与 **反向链接** 保持一直，留意与反向链接的区别。

![image](https://user-images.githubusercontent.com/81074/175893947-46f5d415-7ea0-4aa6-b8ed-56ad92451ba5.png)

**Comment by afenc** [comment]
照 [简悦 +Logseq 个人知识库搭建 | 从零开始完全指南 - 少数派](https://sspai.com/post/72022) 設定順利可用，與這個新解法有很大的差別嗎？
因為我直接設定完後並未得到上述效果…只好復原本來的json。或許應該reset重練？

**Reply by Kenshin** [reply]
嗨，抱歉之前漏掉了这条。 😂

区别在于：Logseq Sync 方案可以 Embed 在 Logseq 中，并且有相关的 Logseq 插件 https://github.com/Kenshin/simpread/discussions/4220

你提到的方式相当于一个轻量级使用方案；而 Logseq SimpRead Sync 属于重度用户使用方案。

**Comment by qqwxp** [comment]
"在任意 http://localhost:7026/unread/2448 加上后缀 ?embed=logseq"

是指在浏览器地址栏输入“http://localhost:7026/unread/2448?embed=logseq” 吗？我在edge浏览器上输入这个地址，显示的是
![image](https://user-images.githubusercontent.com/10953292/188051783-8af41b62-412d-4d0c-afd0-b6160a08e420.png)

详细讨论内容 https://github.com/Kenshin/simpread/issues/4449

**Reply by xiang578** [reply]
需要在 output 目录有一个名字为 2448-xxx.html 的文件

**Reply by xiang578** [reply]
你把这个数字改成一篇你自己有的笔记尝试一下。

**Comment by Kenshin** [comment]
嗨，各位。

简悦官方针对新用户或不喜欢配置的用户，发布了一个新功能：**Logseq 配置库**。

细节请看下面，几分钟就能配置成完成。

***

![img](https://cdn.jsdelivr.net/gh/23784148/upload-images@main/typora/20221007_1665118121.png)

> 针对新用户的一步到位的 Logseq 配置库。

## 🗂 配置库

[简悦 · 配置库](https://www.yuque.com/kenshin/simpread/ds8zk0) 是简悦官方推出的一套针对新用户的极简配置方案，方便新用户用最快的方式使用简悦的各种高级服务，配置库内置了常用的双链笔记用法，如：Notion、Obsidian、Logseq、Roam Research，同时包含了简悦在阅读模式上的一些常规插件：Live Editor、题图、Safari 阅读模式等。

## 💡 特点

用户仅需要下载对应的配置库，修改里面的 UID 为自己的高级账户 UID，即可实现简悦的各种高级操作。

## 💁‍♂️ 受众

新用户或者不想使用手动配置的用户。

## 📝 如何使用

1️⃣ 下载

2️⃣ 修改为自己的 UID

3️⃣ 将必要的文件夹复制粘贴到相应位置

✅ Done！

## 📦 Logseq 配置库

因为 Logseq 与 Obsidian 在插件 API 的支持度有限，所以并没有轻量级使用方案，根据 Logseq 支持的文件格式区别两种方案。

这两种方案均需要同步助手，几乎无任何配置，支持本地快照，自动同步标注，可在 Logseq 中标注，适合 Logseq + 简悦的重度用户。

### 🔌 功能

1️⃣ 支持在 Logseq 中标注

2️⃣ 加入稍后读，自动保存本地快照

3️⃣ 加入标注时，自动将标注内容导入本地快照和 Logseq

4️⃣ 改变稍后读或标注的元数据时，自动将改动的内容保存到本地快照和 Logseq

### 📖 区别

![img](https://imgs.zhubai.love/66078e55f4764b3eaf1a7c1f7cc64c28.png)

### 📙 Org Mode 版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/logseq@orgmode/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/mdgp9n)

### 📗 Markdown 版

🔗 [Github](https://github.com/Kenshin/simpread-configs/blob/main/logseq@md/Getting%20Started.md) | [语雀](https://www.yuque.com/kenshin/simpread/tukkeo)
